prov/gni/src/gnix_msg.c

```diff
@@ -759,6 +908,14 @@ ssize_t _gnix_recv(struct gnix_fid_ep *ep, uint64_t buf, size_t len,
 			_gnix_fr_free(ep, req);
 		}
 	} else {
+		/* if peek/claim/discard, we didn't find what we
+		 * were looking for, return FI_ENOMSG
+		 */
+		if (r_flags) {
+			ret = -FI_ENOMSG;
+			goto pdc_exit;
+		}
```

Comment:
Don't we need to generate an error event in this case?

fi_tagged man page:
" If no message is found matching the tags specified in the peek
       request, then a completion queue error entry with err field set to FI_ENOMSG will be available."